app/models/forms/delete_logs_form.rb

```diff
@@ -0,0 +1,40 @@
+module Forms
+  class DeleteLogsForm
+    include ActiveModel::Model
+    include ActiveModel::Validations
+
+    attr_reader :logs, :log_type, :selected_ids, :search_term, :delete_confirmation_path, :back_to_logs_path, :delete_path
+
+    validate :at_least_one_log_selected
+
+    def initialize(attributes)
+      @log_type = attributes[:log_type]
+      @search_term = attributes[:search_term]
+      @current_user = attributes[:current_user]
+      @logs = FilterService.filter_logs(all_logs, @search_term, attributes[:log_filters], nil, @current_user)
+      @selected_ids = attributes[:selected_ids] || @logs.map(&:id)
```

Comment:
As far as I can see, `attributes[:selected_ids]` will always have values, or default to `[]`. Is that right?

If you fetch with a default then you can remove the defaulting outside of this class:

```suggestion
      @selected_ids = attributes.fetch(:selected_ids, [])
```

Reply:
There may be a better way to organise this:
- when the user initially navigates from an index page, there are no selected_ids in attributes and we wish to create an array with all ids in it (we can only create this array after calling the FilterService, which happens in the Form object initialize)
- if the user presses cancel on the confirmation page, he is returned with an array of selected_ids which are passed into the form object initialize
- if the user unselects all checkboxes and presses delete, then in the controller confirmation methods `params[:forms_delete_logs_form]` has no selected_ids key in it at all, so if it is not present, we set it to be an empty array, eg on line 25.
I can't immediately see a way to rejig what I've done to make it cleaner and account for these situations, but I'm happy to hear suggestions.